pkg/project/util/util_test.go

```diff
@@ -0,0 +1,43 @@
+package util_test
+
+import (
+	"reflect"
+	"testing"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+
+	"github.com/google/gofuzz"
+	"github.com/openshift/origin/pkg/project/api"
+	"github.com/openshift/origin/pkg/project/util"
+)
+
+// TestProjectFidelity makes sure that the project to namespace round trip does not lose any data
+func TestProjectFidelity(t *testing.T) {
+	f := fuzz.New().NilChance(0)
+	p := &api.Project{}
+	for i := 0; i < 1000; i++ {
+		f.Fuzz(p)
+		p.TypeMeta = unversioned.TypeMeta{} // Ignore TypeMeta
+		namespace := util.ConvertProject(p)
+		p2 := util.ConvertNamespace(namespace)
+		if !reflect.DeepEqual(p, p2) {
+			t.Errorf("project data not preserved %#v does not equal %#v", p, p2)
```

Comment:
`t.Log(diff.ObjectDiff(p,p2))` before failing is useful to help see the differences